### PR TITLE
[flang] Add lowering stubs for OpenMP/OpenACC declarative construct in module

### PR DESCRIPTION
**Pull request by PeixinQiao**
The OpenMP/OpenACC declarative construct can be declared in module declaration scope. This patch adds interface for lowering it.

### CODE REVIEW

File: flang/test/Lower/OpenMP/omp-threadprivate.f90 [hunk 1]
```diff
@@ -3,6 +3,11 @@
 ! RUN: %bbc -fopenmp -emit-fir %s -o  - | \
 ! RUN:   FileCheck %s --check-prefix=FIRDialect
```
[the remark this answers is not in the document]
Comment (PeixinQiao) [reply flang/test/Lower/OpenMP/omp-threadprivate.f90:4]: This is only a stub.
```
// Without this patch
$ bbc -fopenmp -emit-fir case.f90 
bbc: /home/.../f18-llvm-project/flang/lib/Lower/PFTBuilder.cpp:369: Fortran::lower::pft::Evaluation& {anonymous}::PFTBuilder::addEvaluation(Fortran::lower::pft::Evaluation&&): Assertion `!evaluationListStack.empty() && "empty evaluation list stack"' failed.
// With this patch
$ bbc -fopenmp -emit-fir case.f90 
error: loc("./case.f90":1:1): /home/.../f18-llvm-project/flang/lib/Lower/OpenMP.cpp:712: not yet implemented OpenMPThreadprivate
LLVM ERROR: aborting
```

**Review by schweitzpgi — CHANGES_REQUESTED** [review]
Conflicts need to be resolved.

**Comment by PeixinQiao** [comment]
@schweitzpgi Thanks for the notice. Addressed the conflicts.

**Review by schweitzpgi — APPROVED** [review]

**Comment by schweitzpgi** [comment]
@kiranchandramohan Are you ok with this one?

**Review by kiranchandramohan — APPROVED** [review]
LGTM.
OK with me @schweitzpgi

**Pull request merged**